src/vs/workbench/contrib/debug/browser/debugEditorActions.ts

```diff
@@ -230,7 +231,29 @@ export class RunToCursorAction extends EditorAction {
 			// otherwise set it at the precise column #102199
 			column = position.column;
 		}
-
+		if (debugService.state === State.Inactive) {
+			// add a break point to be removed later
```

Comment:
This appears to duplicate the logic in `debugService.runTo`, I'd prefer to find a way to reuse that code somehow

Reply:
@roblourens This seems a bit challenging . Run to works under the assumption that the debugger is in stop state.
If I try to add the inactive condition and start the debugger , the temporary breakpoint won't get removed.

Do you have any ideas on how to overcome this ?

Reply:
> If I try to add the inactive condition and start the debugger inside the runTo , the temporary breakpoint won't get removed.

I don't understand the problem

Reply:
Inside the `debuggerService.runTo`, If I add the logic to start the debugger when it is in `INACTIVE` state 
(preferably [here](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/browser/debugService.ts#L1171) )

```ts
let focusedSession = this.getViewModel().focusedSession;
if (this.state === State.Inactive) {
	const { launch, name, getConfig } = this.getConfigurationManager().selectedConfiguration;
	const config = await getConfig();
	const configOrName = config ? Object.assign(deepClone(config), {}) : name;
	await this.startDebugging(launch, configOrName, undefined, true);
	focusedSession = this.getViewModel().focusedSession;
}
else if (this.state !== State.Stopped) {
	return;
}
if (!focusedSession) {
	return;
}
```

, the breakpoint which gets added [here](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/browser/debugService.ts#L1194) (to be removed later) won't get removed . 

```ts
const oneTimeListener = threadToContinue.session.onDidChangeState(() => {
const state = focusedSession?.state;
if (state === State.Stopped || state === State.Inactive) {
if (breakpointToRemove) {
	this.removeBreakpoints(breakpointToRemove.getId());
}
oneTimeListener.dispose();
}
});
```

Isn't that the expected behaviour of the `runTo` action the breakpoint is temp and not permanent .

Reply:
> won't get removed .

Why not? I still don't understand. Seems like I am looking at two methods that have most of their logic in common and I am sure there is a way to share most of it.

Reply:
Well the reason is due to the fact that 'threadToContinue' is 'undefined' after the debugger is started.


I get your suggestion about reusing the logic. However I'm still not sure how to . 

One idea I have is to separate the logic that can be abstracted out of the 'debuggerService.runTo' namely 
1. The logic to add the temp breakPoint 
2. The logic to remove the temp breakpoint

Reply:
Makes sense to separate out some code that can be shared, or rewrite it in a way that you will have a session to listen to in either case.